### PR TITLE
Migrate from assertThat to assertQueryFailure

### PR DESCRIPTION
**Pull request by pvliborio**
Solves #8274 

- Migrate from io.trino.tempto.assertions.QueryAssert#assertThat(QueryCallback) to assertQueryFailure(QueryCallback)

### CODE REVIEW

**Review by hashhar — COMMENTED** [review]
Please take a look at Checkstyle errors.

Also use `hasMessageContaining` instead of `hasMessage`. The `failsWithMessage` checked using `contains` instead of equality.

**Review by pvliborio — COMMENTED** [review]
The .hasMessageMatching doesn't match `java.sql.SQLException: `. Would it be okay to remove that part?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestSyncPartitionMetadata.java [hunk 1]
```diff
@@ -98,8 +99,8 @@ public void testInvalidSyncMode()
         String tableName = "test_repair_invalid_mode";
         prepare(hdfsClient, hdfsDataSourceWriter, tableName);
 
-        assertThat(() -> query("CALL system.sync_partition_metadata('default', '" + tableName + "', 'INVALID')"))
-                .failsWithMessageMatching("java.sql.SQLException: Query failed (.*): Invalid partition metadata sync mode: INVALID");
+        assertQueryFailure(() -> query("CALL system.sync_partition_metadata('default', '" + tableName + "', 'INVALID')"))
+                .hasMessageMatching("java.sql.SQLException: Query failed (.*): Invalid partition metadata sync mode: INVALID");
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestSyncPartitionMetadata.java:103]: should be safe to remove `java.sql.SQLException`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java [hunk 2]
```diff
@@ -1418,9 +1419,9 @@ public void testAcidUpdateWithSubqueryAssignment()
             });
 
             // SET with correlated subquery
-            assertThat(() -> onTrino().executeQuery(format("UPDATE %s SET column2 = (SELECT name FROM tpch.tiny.region WHERE column1 = regionkey)", tableName)))
+            assertQueryFailure(() -> onTrino().executeQuery(format("UPDATE %s SET column2 = (SELECT name FROM tpch.tiny.region WHERE column1 = regionkey)", tableName)))
                     // TODO (https://github.com/trinodb/trino/issues/3325) support correlated UPDATE
-                    .failsWithMessageMatching("\\Qjava.sql.SQLException: Query failed (#\\E\\S+\\Q): Invalid descendant for DeleteNode or UpdateNode: io.trino.sql.planner.plan.MarkDistinctNode");
+                    .hasMessageMatching("\\Qjava.sql.SQLException: Query failed (#\\E\\S+\\Q): Invalid descendant for DeleteNode or UpdateNode: io.trino.sql.planner.plan.MarkDistinctNode");
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java:1424]: should be safe to remove `java.sql.SQLException`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java [hunk 3]
```diff
@@ -1389,9 +1390,9 @@ public void testAcidUpdateWithSubqueryPredicate()
             });
 
             // WHERE with correlated subquery
-            assertThat(() -> onTrino().executeQuery(format("UPDATE %s SET column2 = 'row updated yet again' WHERE column2 = (SELECT name FROM tpch.tiny.region WHERE regionkey = column1)", tableName)))
+            assertQueryFailure(() -> onTrino().executeQuery(format("UPDATE %s SET column2 = 'row updated yet again' WHERE column2 = (SELECT name FROM tpch.tiny.region WHERE regionkey = column1)", tableName)))
                     // TODO (https://github.com/trinodb/trino/issues/3325) support correlated UPDATE
-                    .failsWithMessageMatching("\\Qjava.sql.SQLException: Query failed (#\\E\\S+\\Q): Invalid descendant for DeleteNode or UpdateNode: io.trino.sql.planner.plan.MarkDistinctNode");
+                    .hasMessageMatching("\\Qjava.sql.SQLException: Query failed (#\\E\\S+\\Q): Invalid descendant for DeleteNode or UpdateNode: io.trino.sql.planner.plan.MarkDistinctNode");
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveTransactionalTable.java:1395]: should be safe to remove `java.sql.SQLException`.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaAvroWritesSmokeTest.java [hunk 4]
```diff
@@ -61,11 +62,11 @@ public void testInsertPrimitiveDataType()
...
     public void testInsertStructuralDataType()
     {
-        assertThat(() -> query(format(
+        assertQueryFailure(() -> query(format(
                 "INSERT INTO %s.%s VALUES " +
                         "(ARRAY[100, 102], map_from_entries(ARRAY[('key1', 'value1')]))",
                 KAFKA_CATALOG,
                 STRUCTURAL_AVRO_TABLE_NAME)))
-                .failsWithMessageMatching("java.sql.SQLException: Query failed \\(.+\\): Unsupported column type 'array\\(bigint\\)' for column 'c_array'");
+                .hasMessageMatching("java.sql.SQLException: Query failed \\(.+\\): Unsupported column type 'array\\(bigint\\)' for column 'c_array'");
```
Comment (hashhar) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/kafka/TestKafkaAvroWritesSmokeTest.java:70]: should be safe to remove `java.sql.SQLException`.

**Review by hashhar — APPROVED** [review]
LGTM % ci failures.

It's surprising that the earlier `failsMessageMatching` passed for regexes having exception name in the assertion. AFAIK exception's `message` doesn't include the exception name.

cc: @findepi Do you have any idea why this is behaving like this? The old and new impl look same to me (`str.matches(regex)`)?

**Comment by pvliborio** [comment]
@hashhar @findepi I didn't understand the error in CI. Can you help me?

**Comment by hashhar** [comment]
> @hashhar @findepi I didn't understand the error in CI. Can you help me?

Can you push an empty commit? That failure looks like a flakyness.

`git commit --allow-empty -m 'empty for ci'`

**Comment by pvliborio** [comment]
> > @hashhar @findepi I didn't understand the error in CI. Can you help me?
> 
> Can you push an empty commit? That failure looks like a flakyness.
> 
> `git commit --allow-empty -m 'empty for ci'`

Done.

**Comment by pvliborio** [comment]
> > @hashhar @findepi I didn't understand the error in CI. Can you help me?
> 
> Can you push an empty commit? That failure looks like a flakyness.
> 
> `git commit --allow-empty -m 'empty for ci'`

Worked! Thank you.

**Comment by hashhar** [comment]
Thanks @pvliborio . All tests passed.
Have you sent the signed CLA using the same email you use with your GitHub account/the one you have in your GitHub commit? We require the CLA before we can merge the code.

**Comment by pvliborio** [comment]
> Thanks @pvliborio . All tests passed.
> Have you sent the signed CLA using the same email you use with your GitHub account/the one you have in your GitHub commit? We require the CLA before we can merge the code.

Yes. Sent on 06/21/2021 08:57:34 -0300 to cla@trino.io.

**Comment by hashhar** [comment]
@martint Can you please help with the CLA here? This looks good to go otherwise.

**Comment by pvliborio** [comment]
@hashhar CLA done!

**Comment by hashhar** [comment]
@clabot check

**Comment by hashhar** [comment]
@cla-bot check

**Comment by pvliborio** [comment]
@hashhar I think this resolved, sorry, it was my mistake in the first commit.

**Comment by hashhar** [comment]
Awesome @pvliborio. Can you please squash all the commits into a single one. This is logically a single change so should be kept as such in the history.

Will merge once CI finishes after you push.

**Comment by pvliborio** [comment]
@hashhar Done. Thanks for the help!

**Review by hashhar — APPROVED** [review]

**Pull request merged**